docstore/gcpfirestore/fs.go

```diff
@@ -711,6 +712,28 @@ func withResourceHeader(ctx context.Context, resource string) context.Context {
 	return metadata.NewOutgoingContext(ctx, md)
 }
 
+// RevisionToString implements driver.RevisionToString.
+func (c *collection) RevisionToString(rev interface{}) (string, error) {
+	r, ok := rev.(*tspb.Timestamp)
+	if !ok {
+		return "", gcerr.Newf(gcerr.InvalidArgument, nil, "revision %v is not a  proto Timestamp", rev)
+	}
+	bytes, err := proto.Marshal(r)
+	if err != nil {
+		return "", err
+	}
+	return string(bytes), nil
```

Comment:
Good point, it doesn't. 

I wrote a little server and did an end-to-end experiment. Non-printing bytes round-trip fine (at least with chrome), but a `"` character in the string will break silently: the HTML
```
<input type="hidden" name="foo" value="what"s up doc">
```
will appear to work, but the returned value is `what`.

So we can either:
1. Tell people that the string isn't safe.
2. Use a `[]byte` so they know it has to be encoded.
3. Make it safe (by uuencoding it, for example).

I prefer (3) since it's not hard and provides the best user experience.

Reply:
I like 3 as well.

Reply:
Done, except with base64. See https://retrocomputing.stackexchange.com/questions/3019/why-did-base64-win-against-uuencode/3020.

Reply:
PTAL.